File: sdk/examples/17_create_customer.rs
mod utils;
use testing::USER_SATOSHI;
use utils::init_sdk;

#[allow(clippy::unwrap_used, clippy::expect_used)]
#[tokio::main]
async fn main() {
    // Initialize SDK and create a new user
    let (mut sdk, _cleanup) = init_sdk().await;
    let user: utils::TestUser = (*USER_SATOSHI).clone().into();

    sdk.create_new_user(&user.username).await.unwrap();
    sdk.init_user(&user.username).await.unwrap();

    // Create sap customer if not exists
    if sdk.get_customer().await.is_err() {
        sdk.create_customer("DE").await.unwrap();
    } else {
        println!("Customer already exists")
    }
}

File: sdk/bindings/wasm/examples/webapp/examples/17-create_customer.js
import { initSdk } from './util';
import * as wasm from "cawaena-sdk-wasm";

async function main() {
    let username = "satoshi";
    const country_code = "DE";
    const sdk = await initSdk();
    await sdk.createNewUser(username);
    await sdk.initializeUser(username);
    try {
        await sdk.getCustomer();
        console.log("customer already exists.");
    } catch (error) {
        console.log(error);
        await sdk.createCustomer(country_code);
        console.log("user created successfully");

    }
}

export { main }
